pytest/testsuites/test_xdp.py

```diff
@@ -0,0 +1,37 @@
+"""Runs 'FunctionalTests-XDP.xml' using Pytest."""
+
+
+import conftest
+import pytest
+from node_plugin import Node
+
+
+@pytest.mark.lisa(
+    platform="Azure",
+    category="Functional",
+    area="XDP",
+    tags=["xdp", "network", "hv_netvsc", "sriov"],
+    priority=0,
+)
+@pytest.mark.deploy(
+    setup="OneVM2NIC",
+    networking="SRIOV",
+    vm_image="Canonical:0001-com-ubuntu-server-focal:20_04-lts:latest",
+    vm_size="Standard_DS4_v2",
+)
+@pytest.mark.skip(reason="Not Finished")
+def test_verify_xdp_compliance(node: Node) -> None:
+    for f in [
+        "utils.sh",
+        "XDPDumpSetup.sh",
+        "XDPUtils.sh",
+        "enable_passwordless_root.sh",
+        "enable_root.sh",
+    ]:
+        node.put(conftest.LINUX_SCRIPTS / f)
+        node.run(f"chmod +x {f}")
```

Comment:
Maybe we should revisit how these scripts are copied?  Maybe something like having a `node.put_files([...], mode=...)` or something of the sort?

Reply:
That's a TODO I didn't write down 😅 I was thinking `node.copy_scripts()` that would 1) upload them, 2) make them executable, and 3) delete them when the fixture is torn down.

Reply:
The refactor in my head that I've been putting off because it doesn't add any concepts to be proved is to take `Node` and override it's existing initializer and context manager so make the `node` fixture much cleaner, and would let us easily create a `two_node` fixture (or an arbitrary `environment` fixture) which instantiates multiples nodes for tests which require them.